bot/exts/utils/_eval_helper.py

```diff
@@ -0,0 +1,339 @@
+import re
+import urllib.parse
+import zlib
+from functools import partial
+from io import BytesIO
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import aiohttp
+from discord import Embed
+from discord.ext import commands
+from discord.ext.commands import Context
+from loguru import logger
+
+to_bytes = partial(bytes, encoding="utf-8")
+
+
+def _to_tio_string(couple: tuple) -> Any:
+    """
+    Return a tio api compatible string.
+
+    For example:
+    (language) bash -> b'Vlang\x001\x00bash\x00'
+    (code) echo "bash" -> b'F.code.tio\x0019\x00echo "bash"\n\x00'
+    """
+    name, obj = couple
+    if not obj:
+        return b""
+    elif isinstance(obj, list):
+        content = [f"V{name}", str(len(obj))] + obj
+        return to_bytes("\x00".join(content) + "\x00")
+    else:
+        return to_bytes(f"F{name}\x00{len(to_bytes(obj))}\x00{obj}\x00")
+
+
+class Tio:
+    """Helper class for eval command."""
+
+    def __init__(
+        self,
+        language: str,
+        code: str,
+        inputs: str = "",
+        compiler_flags: Optional[list] = None,
+        command_line_options: Optional[list] = None,
+        args: Optional[list] = None,
+    ):
+        if args is None:
+            args = []
+        if command_line_options is None:
+            command_line_options = []
+        if compiler_flags is None:
+            compiler_flags = []
+        self.backend = "https://tio.run/cgi-bin/run/api/"
+        self.json = "https://tio.run/languages.json"
+
+        strings = {
+            "lang": [language],
+            ".code.tio": code,
+            ".input.tio": inputs,
+            "TIO_CFLAGS": compiler_flags,
+            "TIO_OPTIONS": command_line_options,
+            "args": args,
+        }
+
+        bytes_ = (
+            b"".join(
+                map(
+                    _to_tio_string,  # func
+                    zip(strings.keys(), strings.values()),  # iterables
+                )
+            )
+            + b"R"
+        )
+
+        # This returns a DEFLATE-compressed byte-string, which is what the API requires
+        self.request = zlib.compress(bytes_, 9)[2:-4]
+
+    async def get_result(self) -> Any:
+        """Send Request to Tio Run API And Get Result."""
+        async with aiohttp.ClientSession() as client_session:
+            async with client_session.post(self.backend, data=self.request) as res:
+                if res.status != 200:
+                    logger.warning(
+                        f"HttpProcessingError while getting result of code from tio api with "
+                        f"status code: {res.status}"
+                    )
+
+                data = await res.read()
+                data = data.decode("utf-8")
+                return data.replace(data[:16], "")  # remove token
```

Comment:
`.replace` is a method for strings, so this method shouldn't return `Any`. This should return a `str` object.
```suggestion
    async def get_result(self) -> str:
        """Send Request to Tio Run API And Get Result."""
        async with aiohttp.ClientSession() as client_session:
            async with client_session.post(self.backend, data=self.request) as res:
                if res.status != 200:
                    logger.warning(
                        f"HttpProcessingError while getting result of code from tio api with "
                        f"status code: {res.status}"
                    )

                data = await res.read()
                data = data.decode("utf-8")
                return data.replace(data[:16], "")  # remove token
                ```